limits/storage.py

```diff
@@ -221,6 +228,14 @@ def check(self):
         """
         return True
 
+    def reset(self):
+        self.timer.cancel()
+        self.storage = Counter()
```

Comment:
Instead of creating new counters/dictionaries could you call `clear` on all of them?